src/io.rs

```diff
@@ -392,3 +392,16 @@ impl<T: AsyncWrite, U: Encoder> FramedWrite<T, U> {
         self.inner.0.borrow().handle
     }
 }
+
+impl<T: AsyncWrite, U: Encoder> Drop for FramedWrite<T, U> {
+    fn drop(&mut self) {
+        // Attempts to write any remaining bytes to the stream and flush it
+        let mut async_writer = self.inner.1.borrow_mut();
+        let inner = self.inner.0.borrow_mut();
+        if !inner.buffer.is_empty() {
+            // Results must be ignored during drop
```

Comment:
Comments should explain why the code is done this way, not to just state fact :)

Please state that we cannot possibly handle errors during drop, so if it is not possible to write remaining data, then we drop it.

Reply:
Changed a bit